### PR TITLE
content(performance): don't call functions from the template to get data

### PR DESCRIPTION
**Pull request by MaciejWWojcik**
Add new content in the performance category about performance issues as a result of calling functions directly from the template

### CODE REVIEW

**Review by PooSham — CHANGES_REQUESTED** [review]
You seem to have forgotten to add `changeDetection: ChangeDetectionStrategy.OnPush` in the "Using OnPush change Detection strategy" section. 

Also, we should probably change `components/minimize-logic-in-templates.md` too, since it gives the bad advice to do calls from the tempalte

**Comment by MaciejWWojcik** [comment]
Oh, thanks! I updated the pull request with missing ChangeDetection line :)

File: content/performance/dont-call-function-from-template.md [hunk 1]
```diff
@@ -0,0 +1,131 @@
...
+
+  getName() {
+    return this.service.getName(this.id);
+  }
+}
+```
+
+That's why Angular will know the exact value of the property all the time, without any cost of checking that on every Change Detection!
+
+## Using OnPush Change Detection strategy
```
Comment (d3lm) [inline content/performance/dont-call-function-from-template.md:91]:
```suggestion
## Using OnPush change detection strategy
```

File: content/performance/dont-call-function-from-template.md [hunk 2]
```diff
@@ -0,0 +1,131 @@
...
+Change Detection strategy describes how Angular should handle Change Detection and DOM rendering. When using OnPush strategy, the function will be called again only if `@Input` properties changes, or if `@Output` emiters are fired or an `Observable` emits new values.
```
[the remark this answers is not in the document]
Comment (d3lm) [reply content/performance/dont-call-function-from-template.md:92]:
```suggestion
The change detection strategy describes how Angular should handle change detection. When using the `OnPush` strategy, change detection only runs if an `@Input` property changes, event are emitted e.g. an `@Output`, or an `Observable` emits new values. Furthermore, change detection can be executed manually via the `ChangeDetectorRef`.
```

File: content/performance/dont-call-function-from-template.md [hunk 3]
```diff
@@ -0,0 +1,131 @@
...
+  @Input() id;
+
+  getName() {
+    return this.service.getName(this.id);
+  }
+}
+```
+
+## Using custom pipes
+Pipes by default are pure, which means Angular will execute its `transform` method only on change to the input value.
```
Comment (d3lm) [inline content/performance/dont-call-function-from-template.md:111]: Newline before.

File: content/performance/dont-call-function-from-template.md (hunk 2 above)
[the remark this answers is not in the document]
Comment (d3lm) [reply content/performance/dont-call-function-from-template.md:92]: See https://angular.io/api/core/ChangeDetectorRef#!#detectChanges-anchor

File: content/performance/dont-call-function-from-template.md [hunk 4]
```diff
@@ -0,0 +1,131 @@
...
+Given that example:
```
Comment (d3lm) [inline content/performance/dont-call-function-from-template.md:112]: Newline before, and:

```suggestion
Here's an example:
```

File: content/performance/dont-call-function-from-template.md [hunk 5]
```diff
@@ -0,0 +1,131 @@
...
+  template: `
+    <div>{{ id | getName }}</div>
+  `
+})
+export class MyComponent {
+  id;
+}
+```
+
+As long as the `id` property will stay the same, the pipe won't execute its code again and perform a possibly expensive task.
```
Comment (d3lm) [inline content/performance/dont-call-function-from-template.md:125]:
```suggestion
As long as the `id` property doesn't change the pipe won't execute its code again and perform a possibly expensive task.
```

File: content/performance/dont-call-function-from-template.md [hunk 6]
```diff
@@ -0,0 +1,131 @@
...
+
```
Comment (d3lm) [inline content/performance/dont-call-function-from-template.md:126]: Remove one newline.

**Review by d3lm — CHANGES_REQUESTED** [review]
Love it! Thanks for the PR 🙏 Left some remarks.

**Comment by MaciejWWojcik** [comment]
Thank you for the remarks and suggestions! Now it looks a way better 😄 🙏

**Review by d3lm — APPROVED** [review]

**Comment by d3lm** [comment]
Good job! Thanks for the PR 🙏

**Pull request merged**